api/v1alpha1/api.go

```diff
@@ -1,3 +1,118 @@
 package v1alpha1
 
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	gwapiv1a2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
+)
+
+// +kubebuilder:object:root=true
+
+// LLMRoute combines multiple LLMBackends and attaching them to Gateway(s) resources.
+//
+// This serves as a way to define a "unified" LLM API for a Gateway which allows downstream
+// clients to use a single schema API to interact with multiple LLM backends.
+//
+// The InputSchema is used to determine the structure of the requests that the Gateway will
+// receive. And then the Gateway will route the traffic to the appropriate LLMBackend based
+// on the output schema of the LLMBackend while doing the other necessary jobs like rate limit, etc.
+type LLMRoute struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	// Spec defines the details of the LLM policy.
+	Spec LLMRouteSpec `json:"spec,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+
+// LLMRouteList contains a list of LLMTrafficPolicy
+type LLMRouteList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []LLMRoute `json:"items"`
+}
+
+// LLMRouteSpec details the LLMRoute configuration.
+type LLMRouteSpec struct {
+	// APISchema specifies the API schema of the input that the target Gateway(s) will receive.
+	// Based on this schema, the ai-gateway will perform the necessary transformation to the
+	// output schema specified in the selected LLMBackend during the routing process.
+	APISchema LLMAPISchema `json:"inputSchema"`
```

Comment:
```suggestion
	APISchema LLMAPISchema `json:"apiSchema"`
```

Reply:
Are we going to allow different APISchema for different LLM route if we define at the route level?

Reply:
If my understanding is correct, this indicates which vendor and model the route belongs to. Therefore, a single LLMRoute can correspond to only one vendor and model

Reply:
> Are we going to allow different APISchema for different LLM route if we define at the route level?

I think the answer would be no at the moment since i am not sure why a user wants to access API like that - clients need to use different set of API clients and switching it from their end depending on a path? idk.